### PR TITLE
Feature master 920 forward files in received shares

### PR DESCRIPTION
**Pull request by ppppbn**

### CODE REVIEW

File: app/modules/linshare.components/sidebar/sidebarQuickShare.html [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+                {{"NAVIGATION.CANCEL" | translate}}
+              </button>
+              <button class="btn btn-primary col-xs-6">
+                {{"NAVIGATION.SEND" | translate}}
+              </button>
+            </div>
+          </div>
+        </div>
+      </div>
+      <form role="form" class="clearfix">
```
Comment (RockyMeadow) [inline app/modules/linshare.components/sidebar/sidebarQuickShare.html]: no need another form here, entire component should be wrapped in a single form

File: app/modules/linshare.components/sidebar/sidebarQuickShare.html [hunk 2]
```diff
@@ -0,0 +1,269 @@
+<div class="card col-xs-12 form-within-sidebar create-share scrollable-sidebar-area" id="content-ctn-sidebar">
```
Comment (RockyMeadow) [inline app/modules/linshare.components/sidebar/sidebarQuickShare.html]: this component should be put in linshare.share module, you can limit the templates to only parts required for the form

File: app/modules/linshare.components/sidebar/sidebarQuickShare.html [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        ng-click="sidebarQuickShareVm.toggleDisplayAdvancedOptions()">
+        <a href="" translate>
+          RIGHT_PANEL.SHARE.MORE_OPTIONS
+        </a>
+        <i class="zmdi zmdi-chevron-down toggle-chevron m-l-5"></i>
+      </div>
+
+      <div class="advanced-options p-l-5 p-r-5" id="advanced-options-container" ng-class="{ 'advanced-options-visible': sidebarQuickShareVm.displayAdvancedOptions}">
+        <div class="form-group m-b-20">
+          <label class="m-b-10 label-muted">{{"RIGHT_PANEL.SHARE_OPTIONS.ADD_NOTE" | translate}} :</label>
```
Comment (RockyMeadow) [inline app/modules/linshare.components/sidebar/sidebarQuickShare.html]: use translate directive

File: app/modules/linshare.components/sidebar/sidebarQuickShare.html (hunk 1 above)
Comment (ppppbn) [inline app/modules/linshare.components/sidebar/sidebarQuickShare.html]: remove this form

File: app/modules/linshare.components/sidebar/sidebarQuickShareComponent.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+angular
+  .module('linshare.components')
+  .component('sidebarQuickShare', {
+    template: require('./sidebarQuickShare.html'),
+    controller: 'sidebarQuickShareController',
+    controllerAs: 'sidebarQuickShareVm',
+    bindings: {
+      shareObject: '=',
+      loadSidebarContent: '=',
```
Comment (RockyMeadow) [inline app/modules/linshare.components/sidebar/sidebarQuickShareComponent.js]: what is this for?

File: app/modules/linshare.components/sidebar/sidebarQuickShareController.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+      $('#content-ctn-sidebar').animate(
+        {
+          scrollTop: $('#advanced-options-anchor').offset().top
+        },
+        400
+      );
+    }
+  }
+
+  function submitForward (shareCreationDto, selectedDocuments, form) {
```
Comment (RockyMeadow) [inline app/modules/linshare.components/sidebar/sidebarQuickShareController.js]: if we have one single form, this form can be bound to controller and will be accessed from there

File: app/modules/linshare.components/sidebar/sidebarQuickShareController.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+    if (handleErrors(shareCreationDto, selectedDocuments) || (form && !form.$valid)) {
+      return;
+    }
+
+    return copyItemsToMySpace(selectedDocuments).then(data => {
```
Comment (RockyMeadow) [inline app/modules/linshare.components/sidebar/sidebarQuickShareController.js]: you can spread data variable here

File: app/modules/linshare.receivedShare/receivedShareController.js [hunk 7]
```diff
@@ -94,21 +91,19 @@
 
         $q
           .all([
-            loadTable(),
+            launchTableParamsInitiation(),
```
Comment (RockyMeadow) [inline app/modules/linshare.receivedShare/receivedShareController.js]: you can further refactor this by putting all these in $onInit hook

File: app/modules/linshare.receivedShare/receivedShareController.js [hunk 8]
```diff
@@ -127,19 +122,6 @@
         });
       }
 
-      /**
-       * @name addSelectedDocument
-       * @desc add contacts to list of new contacts to create
-       * @param {Object} document - document to add to the list of selected documents
-       * @memberOf LinShare.receivedShare.ReceivedController
-       */
-      function addSelectedDocument(document) {
```
Comment (RockyMeadow) [inline app/modules/linshare.receivedShare/receivedShareController.js:136]: this is not used anymore?

File: app/modules/linshare.receivedShare/receivedShareController.js [hunk 9]
```diff
@@ -283,48 +265,23 @@
...
+            $scope.lengthOfSelectedDocuments = tableParamsService.lengthOfSelectedDocuments;
+            $scope.resetSelectedDocuments = tableParamsService.resetSelectedItems;
+            $scope.selectedDocuments = tableParamsService.getSelectedItemsList();
+            $scope.selectDocumentsOnCurrentPage = tableParamsService.tableSelectAll;
+            $scope.addSelectedDocument = tableParamsService.toggleItemSelection;
+            $scope.sortDropdownSetActive = tableParamsService.tableSort;
+            $scope.toggleFilterBySelectedFiles = tableParamsService.isolateSelection;
+            $scope.flagsOnSelectedPages = tableParamsService.getFlagsOnSelectedPages();
+            $scope.toggleSelectedSort = tableParamsService.getToggleSelectedSort();
+            // $scope.reloadDocuments = reloadDocuments;
```
Comment (RockyMeadow) [inline app/modules/linshare.receivedShare/receivedShareController.js]: remove comment

File: app/modules/linshare.receivedShare/receivedShareController.js [hunk 10]
```diff
@@ -710,5 +563,26 @@
...
+
+          if (index === -1) {
+            document.isSelected = true;
+            $scope.selectedDocuments.push(document);
+          } else {
+            $log.debug('addItem - item is already in the list');
+          }
+        }
+
+        $scope.loadSidebarContent(lsAppConfig.forward);
```
Comment (RockyMeadow) [inline app/modules/linshare.receivedShare/receivedShareController.js]:
```suggestion
        loadSidebarContent(lsAppConfig.forward);
```

File: app/modules/linshare.receivedShare/views/list.html [hunk 11]
```diff
@@ -210,6 +210,13 @@
                     <span translate>ACTION.COPY_IN_WORKGROUP</span>
                   </a>
                 </li>
+                <li class="divider"></li>
+                <li data-ng-click="onForward()">
```
Comment (RockyMeadow) [inline app/modules/linshare.receivedShare/views/list.html]:
```suggestion
                <li ng-click="onForward()">
```

File: app/modules/linshare.share/services/shareObjectService.js [hunk 12]
```diff
@@ -246,6 +246,19 @@ angular.module('linshare.share')
       return deferred.promise;
     };
 
+    ShareObjectForm.prototype.shareWithNoToast = function() {
```
Comment (RockyMeadow) [inline app/modules/linshare.share/services/shareObjectService.js]: multiple duplicated codes, better refactor share function instead

File: app/modules/linshare.components/sidebar/sidebarQuickShare.html (hunk 3 above)
Comment (ppppbn) [reply app/modules/linshare.components/sidebar/sidebarQuickShare.html]: this has a `:` after the translated text

File: app/modules/linshare.receivedShare/receivedShareController.js (hunk 8 above)
Comment (ppppbn) [reply app/modules/linshare.receivedShare/receivedShareController.js:136]: yes, replaced it with `tableParamsService.toggleItemSelection`

File: app/modules/linshare.components/sidebar/sidebarQuickShare.html (hunk 3 above)
Comment (RockyMeadow) [reply app/modules/linshare.components/sidebar/sidebarQuickShare.html]: you can add `:` on the translation text

File: app/modules/linshare.receivedShare/receivedShareController.js [hunk 13]
```diff
@@ -13,46 +13,36 @@
   // TODO: Should dispatch some function to other service or controller in order to valid the maxparams linter
   /* jshint maxparams: false, maxstatements: false */
```
Comment (RockyMeadow) [inline app/modules/linshare.receivedShare/receivedShareController.js]: remove jshint comment

File: app/modules/linshare.receivedShare/receivedShareController.js [hunk 14]
```diff
@@ -94,21 +84,19 @@
 
         $q
           .all([
-            loadTable(),
+            launchTableParamsInitiation(),
             authenticationRestService.getCurrentUser(),
           ])
           .then(function(promises) {
```
Comment (RockyMeadow) [inline app/modules/linshare.receivedShare/receivedShareController.js]: arrow function please

File: app/modules/linshare.receivedShare/receivedShareController.js [hunk 15]
```diff
@@ -94,21 +84,19 @@
 
         $q
           .all([
-            loadTable(),
+            launchTableParamsInitiation(),
```
Comment (RockyMeadow) [inline app/modules/linshare.receivedShare/receivedShareController.js]: launchTableParamsInitiation does not return a promise, this is useless

File: app/modules/linshare.receivedShare/receivedShareController.js [hunk 16]
```diff
@@ -283,48 +258,22 @@
...
+            $scope.lengthOfSelectedDocuments = tableParamsService.lengthOfSelectedDocuments;
+            $scope.resetSelectedDocuments = tableParamsService.resetSelectedItems;
+            $scope.selectedDocuments = tableParamsService.getSelectedItemsList();
+            $scope.selectDocumentsOnCurrentPage = tableParamsService.tableSelectAll;
+            $scope.addSelectedDocument = tableParamsService.toggleItemSelection;
+            $scope.sortDropdownSetActive = tableParamsService.tableSort;
+            $scope.toggleFilterBySelectedFiles = tableParamsService.isolateSelection;
+            $scope.flagsOnSelectedPages = tableParamsService.getFlagsOnSelectedPages();
+            $scope.toggleSelectedSort = tableParamsService.getToggleSelectedSort();
+
```
Comment (RockyMeadow) [inline app/modules/linshare.receivedShare/receivedShareController.js]: remove blank line

File: app/modules/linshare.receivedShare/receivedShareController.js [hunk 17]
```diff
@@ -710,5 +555,26 @@
...
+      /**
+       * @name onForward
+       * @desc open sidebar content for forwarding files
+       * @param {object} documentFile - document
+       * @memberOf LinShare.receivedShare.receivedShareController
+       */
+      function onForward(document) {
+        if (document) {
+          var index = $scope.selectedDocuments.indexOf(document);
```
Comment (RockyMeadow) [inline app/modules/linshare.receivedShare/receivedShareController.js]:
```suggestion
          const index = $scope.selectedDocuments.indexOf(document);
```

File: app/modules/linshare.share/components/quickShare.html [hunk 18]
```diff
@@ -0,0 +1,268 @@
+<div class="card col-xs-12 form-within-sidebar create-share scrollable-sidebar-area" id="content-ctn-sidebar">
+  <md-content id="sidebar-forward" class='p-b-30'>
+    <form name="quickShareVm.form" class="form-sidebar">
+      <div class="custom-list-container clearfix">
+        <div class="recipients-form-sections clearfix">
+          <h2 class="pull-left headerSubCtn">
+            {{"RIGHT_PANEL.SHARE.RECIPIENTS" | translate}}</h2><span class="required-field-hint">*</span>
```
Comment (RockyMeadow) [inline app/modules/linshare.share/components/quickShare.html]: translate directive

File: app/modules/linshare.share/components/quickShare.html [hunk 19]
```diff
@@ -0,0 +1,268 @@
...
+          <div x-ng-if="!quickShareVm.linshareModeProduction"
+              class="dropdown dropdown-standard popOverForm savelistBtn" uib-dropdown is-open="false"
+              on-toggle="toggled()">
+            <a href="" class="dropdown-toggle sidebar-link" uib-dropdown-toggle x-ng-disabled="disabled"
+              alt="{{'RIGHT_PANEL.SHARE.CREATE_MAILING_LIST_ALT' | translate}}">
+              {{"RIGHT_PANEL.SHARE.CREATE_MAILING_LIST" | translate}}
```
Comment (RockyMeadow) [inline app/modules/linshare.share/components/quickShare.html]: translate directive

File: app/modules/linshare.share/components/quickShare.html [hunk 20]
```diff
@@ -0,0 +1,268 @@
+<div class="card col-xs-12 form-within-sidebar create-share scrollable-sidebar-area" id="content-ctn-sidebar">
+  <md-content id="sidebar-forward" class='p-b-30'>
+    <form name="quickShareVm.form" class="form-sidebar">
+      <div class="custom-list-container clearfix">
+        <div class="recipients-form-sections clearfix">
+          <h2 class="pull-left headerSubCtn">
+            {{"RIGHT_PANEL.SHARE.RECIPIENTS" | translate}}</h2><span class="required-field-hint">*</span>
+          <div x-ng-if="!quickShareVm.linshareModeProduction"
```
Comment (RockyMeadow) [inline app/modules/linshare.share/components/quickShare.html:10]: would be nice if you separate this section by comments for future review

File: app/modules/linshare.share/components/quickShareComponent.js [hunk 21]
```diff
@@ -0,0 +1,11 @@
+angular
+  .module('linshare.components')
+  .component('quickShare', {
+    template: require('./quickShare.html'),
+    controller: 'quickShareController',
+    controllerAs: 'quickShareVm',
+    bindings: {
+      shareObject: '=',
+      cleanUpAfterShare: '='
```
Comment (RockyMeadow) [inline app/modules/linshare.share/components/quickShareComponent.js]:
```suggestion
      onAfterShare: '&'
```

File: app/modules/linshare.share/components/quickShare.html [hunk 22]
```diff
@@ -0,0 +1,268 @@
...
+            <div class="dropdown-menu pull-right">
+              <div class="stop-propagate">
+                <p class="labelListCtn">
+                  <input type="email" class="form-control input-sm" id="labelList"
+                        placeholder="Saisissez le nom de votre liste">
+                </p>
+                <md-divider></md-divider>
+              </div>
+              <div>
+                <button class="btn btn-default col-xs-6">
```
Comment (RockyMeadow) [inline app/modules/linshare.share/components/quickShare.html]: translate directive

File: app/modules/linshare.share/components/quickShare.html [hunk 23]
```diff
@@ -0,0 +1,268 @@
...
+                  {{"NAVIGATION.CANCEL" | translate}}
+                </button>
+                <button class="btn btn-primary col-xs-6">
+                  {{"NAVIGATION.SEND" | translate}}
```
Comment (RockyMeadow) [inline app/modules/linshare.share/components/quickShare.html]: translate directive

File: app/modules/linshare.components/sidebar/sidebarQuickShare.html (hunk 3 above)
Comment (ppppbn) [reply app/modules/linshare.components/sidebar/sidebarQuickShare.html]: why so many works? Note that the placeholder text of the textarea below will also be changed if you add `:` on the translation text

File: app/scripts/directives/sidebar-content/sidebar-content-forward.html [hunk 24]
```diff
@@ -0,0 +1,62 @@
...
+  <a class="pull-right sidebar-link hidden-xs" href=""
+     x-ng-click="activeBtnShowSelection = !activeBtnShowSelection; mainVm.sidebar.getData().toggleFilterBySelectedFiles()"
+     x-ng-show="activeBtnShowSelection">
+    {{"RIGHT_PANEL.SHARE.UNHIDE_FILES" | translate}}
+  </a>
+</div>
+
+<quick-share
+  share-object="shareActionVm.newShare"
+  load-sidebar-content="mainVm.sidebar.getData().loadSidebarContent"
```
Comment (RockyMeadow) [inline app/scripts/directives/sidebar-content/sidebar-content-forward.html]: no longer used

File: app/scripts/directives/sidebar-content/sidebar-content-share.html [hunk 25]
```diff
@@ -41,144 +41,12 @@ <h2 class="header-more-options" translate>RIGHT_PANEL.DETAILS.QUICK_SHARE</h2>
...
-        {{"RIGHT_PANEL.SHARE.SETTINGS" | translate}}
-      </h2>
-      <a class="sidebar-link right" href=""
-         x-ng-click="mainVm.sidebar.getData().loadSidebarContent(mainVm.sidebar.getData().lsAppConfig.moreOptions)">
-        {{"RIGHT_PANEL.SHARE.MORE_OPTIONS" | translate}}
-      </a>
-    </div>
+<quick-share
+  share-object="shareActionVm.newShare"
+  load-sidebar-content="mainVm.sidebar.getData().loadSidebarContent"
```
Comment (RockyMeadow) [inline app/scripts/directives/sidebar-content/sidebar-content-share.html]: no longer used

File: app/modules/linshare.share/components/quickShareComponent.js [hunk 26]
```diff
@@ -0,0 +1,11 @@
+angular
+  .module('linshare.components')
+  .component('quickShare', {
+    template: require('./quickShare.html'),
+    controller: 'quickShareController',
+    controllerAs: 'quickShareVm',
+    bindings: {
+      shareObject: '=',
+      onAfterShare: '='
```
Comment (RockyMeadow) [inline app/modules/linshare.share/components/quickShareComponent.js:9]:
```suggestion
      onAfterShare: '&'
```

Comment (ppppbn) [reply app/modules/linshare.share/components/quickShareComponent.js:9]: Changing the binding like this make the function not work

**Pull request closed without merging**